native/dsterf.go

```diff
@@ -0,0 +1,292 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"math"
+
+	"github.com/gonum/lapack"
+)
+
+// Dsterf computes all eigenvalues of a symmetric tridiagonal matrix using the
+// Pal-Walker-Kahan variant of the QL or QR algorithm.
+//
+// d contains the diagonal elements of the tridiagonal on entry, and contains the
+// eigenvalues in ascending order on exit. d must have length at least n, or
+// Dsterf will panic.
+//
+// e contains the off-digonal elements of the tridiagonal on entry, and is
+// overwritten during the call to Dsterf.
+func (impl Implementation) Dsterf(n int, d, e []float64) (ok bool) {
+	if n < 0 {
+		panic(nLT0)
+	}
+	if n == 0 {
+		return true
+	}
+	if len(d) < n {
+		panic(badD)
+	}
+	if len(e) < n-1 {
+		panic(badE)
+	}
+
+	const (
+		none = 0 // no scaling
+		down = 1 // Scale the values below ssfmax threshhold
+		up   = 2 // Scale the values below ssfmin threshhold
+	)
+
+	// Determine the unit roundoff for this environment.
+	eps := dlamchE
+	eps2 := eps * eps
+	safmin := dlamchS
+	safmax := 1 / safmin
+	ssfmax := math.Sqrt(safmax) / 3
+	ssfmin := math.Sqrt(safmin) / eps2
+
+	// Compute the eigenvalues of the tridiagonal matrix.
```

Comment:
This comment reflects the comment in the fortran. It makes it easier to navigate if it exists.